flake.nix

```diff
@@ -2,177 +2,157 @@
+    nil-crypto3 = {
       url = "https://github.com/NilFoundation/crypto3";
       type = "git";
       submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
     };
-    nil_zkllvm_blueprint = {
+    nil-zkllvm-blueprint = {
       url = "https://github.com/NilFoundation/zkllvm-blueprint";
       type = "git";
       submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        nil_crypto3.follows = "nil-crypto3";
+      };
     };
-    intx = { url = "github:chfast/intx"; flake = false; };
-    hashtree = { url = "github:prysmaticlabs/hashtree"; flake = false; };
-    sszpp = { url = "github:OffchainLabs/sszpp"; flake = false; };
-    evmc = { url = "github:ethereum/evmc"; flake = false; };
-    valijson = {url = "github:tristanpenman/valijson"; flake = false; };
   };
 
-  outputs = { self, nixpkgs, nil_evm_assigner, nil_crypto3, nil_zkllvm_blueprint, ... } @ repos:
+  outputs =
+    { self
+    , nixpkgs
+    , flake-utils
+    , nix-3rdparty
+    , nil-evm-assigner
+    , nil-crypto3
+    , nil-zkllvm-blueprint
+    }:
+    flake-utils.lib.eachDefaultSystem (system:
     let
-      supportedSystems = [
-        "x86_64-linux"
+      pkgs = import nixpkgs {
+        overlays = [ nix-3rdparty.overlays.${system}.default ];
+        inherit system;
+      };
+      evm_assigner = nil-evm-assigner.packages.${system}.default;
+      crypto3 = nil-crypto3.packages.${system}.default;
+      blueprint = nil-zkllvm-blueprint.packages.${system}.default;
+
+      # Default env will bring us GCC 13 as default compiler
+      stdenv = pkgs.stdenv;
+
+      defaultNativeBuildInputs = [
+        pkgs.cmake
+        pkgs.ninja
       ];
 
-      # For all supported systems call `f` providing system pkgs
-      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f {
-        pkgs = import nixpkgs {
-          inherit system;
-        };
+      defaultBuildInputs = [
+        # Default nixpkgs packages
+        pkgs.boost
+        pkgs.valijson
+        # Packages from nix-3rdparty
+        pkgs.sszpp
+        pkgs.evmc
+        # Repo dependencies
+        evm_assigner
+        crypto3
+        blueprint
+      ];
+
+      defaultCheckInputs = [
+        pkgs.gtest
+      ];
+
+      defaultDevTools = [
+        pkgs.doxygen
+        pkgs.clang_17 # clang-format and clang-tidy
+      ];
+
+      releaseBuild = stdenv.mkDerivation {
+        name = "zkEVM-framework";
+
+        nativeBuildInputs = defaultNativeBuildInputs;
+
+        buildInputs = defaultBuildInputs;
+
+        src = self;
+
+        cmakeBuildType = "Release";
+
+        doCheck = false;
+      };
+
+      # TODO: we need to propagate debug mode to dependencies here:
+      # Tracking issue: https://github.com/NilFoundation/zkEVM-framework/issues/58
+      debugBuild = releaseBuild.overrideAttrs (finalAttrs: previousAttrs: {
+        name = previousAttrs.name + "-debug";
+
+        cmakeBuildType = "Debug";
```

Comment:
`dontStrip = true;` should be added here too.

Reply:
Yep, there are a lot of things should be done about debug builds. I decided to move them to separate issue, because they will change the code anyway.

Reply:
- #58